vision_agent/tools/planner_tools.py

```diff
@@ -0,0 +1,298 @@
+import logging
+import shutil
+import tempfile
+from typing import Any, Callable, Dict, List, Optional, Tuple, cast
+
+import numpy as np
+from PIL import Image
+
+import vision_agent.tools as T
+from vision_agent.agent.agent_utils import (
+    DefaultImports,
+    extract_code,
+    extract_json,
+    extract_tag,
+)
+from vision_agent.agent.vision_agent_planner_prompts_v2 import (
+    CATEGORIZE_TOOL_REQUEST,
+    FINALIZE_PLAN,
+    PICK_TOOL,
+    TEST_TOOLS,
+    TEST_TOOLS_EXAMPLE1,
+    TEST_TOOLS_EXAMPLE2,
+)
+from vision_agent.lmm import AnthropicLMM
+from vision_agent.utils.execute import CodeInterpreterFactory
+from vision_agent.utils.image_utils import convert_to_b64
+from vision_agent.utils.sim import load_cached_sim
+
+TOOL_FUNCTIONS = {tool.__name__: tool for tool in T.TOOLS}
+TOOL_RECOMMENDER = load_cached_sim(T.TOOLS_DF)
+
+_LOGGER = logging.getLogger(__name__)
+EXAMPLES = f"\n{TEST_TOOLS_EXAMPLE1}\n{TEST_TOOLS_EXAMPLE2}\n"
+
+
+def extract_tool_info(
+    tool_choice_context: Dict[str, Any]
+) -> Tuple[Optional[Callable], str, str, str]:
+    tool_thoughts = tool_choice_context.get("thoughts", "")
+    tool_docstring = ""
+    tool = tool_choice_context.get("best_tool", None)
+    if tool in TOOL_FUNCTIONS:
+        tool = TOOL_FUNCTIONS[tool]
+        tool_docstring = T.TOOLS_INFO[tool.__name__]
+
+    return tool, tool_thoughts, tool_docstring, ""
+
+
+def extract_tool_info_v2(
+    tool_choice_context: Dict[str, Any]
+) -> Tuple[Optional[Callable], str, str, str]:
+    """Use PICK_TOOL_V2 prompt to extract tool information."""
+    tool_list_priors = {
+        "owl_v2_image": 0.6,
+        "owl_v2_video": 0.6,
+        "ocr": 0.8,
+        "clip": 0.6,
+        "vit_image_classification": 0.5,
+        "vit_nsfw_classification": 0.5,
+        "countgd_counting": 0.9,
+        "florence2_ocr": 0.8,
+        "florence2_sam2_image": 0.6,
+        "florence2_sam2_video_tracking": 0.8,
+        "florence2_phrase_grounding": 0.6,
+        "claude35_text_extraction": 0.9,
+        "detr_segmentation": 0.5,
+        "depth_anything_v2": 0.6,
+        "generate_pose_image": 0.5,
+    }
```

Comment:
@dillonalaird couldn't it be better to create an enum with the list of all the tools and then reuse it here and in the tools.py file? Also, this `tool_list_priors` should be a constant and we should move it out of this function in case we may need it somewhere else.

Reply:
Also, I think this object might be better be stored in the tools.py file for us not to lose a reference to it.

Reply:
This is actually from a failed previous experiment. I will remove this code